crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3161,6 +3161,126 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn not_none_literal() -> anyhow::Result<()> {
+        let mut db = setup_db();
+
+        db.write_file(
+            "src/a.py",
+            r#"
+            a = not None
+            b = not not None
+            "#,
+        )?;
+        assert_public_ty(&db, "src/a.py", "a", "Literal[True]");
+        assert_public_ty(&db, "src/a.py", "b", "Literal[False]");
+
+        Ok(())
+    }
+
+    #[test]
+    fn not_function() -> anyhow::Result<()> {
+        let mut db = setup_db();
+
+        db.write_file(
+            "src/a.py",
+            r#"
+            from typing import reveal_type
+            def f():
+                return 1
+
+            a = not f
+            b = not reveal_type
+            "#,
+        )?;
+
+        assert_public_ty(&db, "src/a.py", "a", "Literal[False]");
+        assert_public_ty(&db, "src/a.py", "b", "Literal[False]");
```

Comment:
The test for `b` fails. Not sure why the returned type is `bool`. Any help here would be appreciated